### PR TITLE
Tick() can handle checkboxes that have no value

### LINKED ISSUE #192: tick() requires a value to be set
jordan.m...@gmail.com reported on Apr 16, 2011 

<pre>
Steps to reproduce:
1. Have a form with a malformed checkbox input (with no value), such as &lt;input type=&quot;checkbox&quot; name=&quot;scene&quot; id=&quot;scene&quot;&gt;
2. Issue a $mech-&gt;tick(&#39;scene&#39;, &#39;&#39;);

Expected output: checkbox is ticked
Resulting output: checkbox is not ticked

This applies to most recent version of WWW::Mechanize on any operating system.
</pre>

### Details

Imported from [Google Code issue 203](http://code.google.com/p/www-mechanize/issues/detail?id=203) via [archive](https://web.archive.org/web/20150227111857/http://code.google.com/p/www-mechanize/issues/detail?id=203)

* **Type**:      Defect
* **Date**:      Apr 16, 2011
* **Reporter**:  jordan.m...@gmail.com
* **Owner**:     ----
* **Priority**:  Medium
* **Status**:    New
* **Labels**:    `WM`

### Comments
**jordan.m...@gmail.com** commented on Apr 16, 2011 :

<pre>
A patch has been provided via Github.
</pre>

**petda...@gmail.com** commented on Apr 16, 2011 :

<pre>
To what repo?  Mech is not yet hosted on Github.
</pre>

**jordan.m...@gmail.com** commented on Apr 16, 2011 :

<pre>
Sorry, the patch is attached.
</pre>

**petda...@gmail.com** commented on Apr 24, 2011 :

<pre>
(No comment was entered for this change.)
</pre>

* Labels : `WM`

### PR DESCRIPTION
**Pull request by simbabque**
I have rebased the work done by @jmadler in #1 and added tests.

Closes #1.
Fixes #192.

### CODE REVIEW

File: t/tick.html [hunk 1]
```diff
@@ -4,9 +4,12 @@
 
 <input type="checkbox" name="foo" value="hello" /> Hello<br />
 <input type="checkbox" name="foo" value="bye"   /> Bye<br />
-<input type="checkbox" name="foo" value="arse" /> Arse<br />
+<input type="checkbox" name="foo" value="parse" /> Parse<br />
```
Comment (oalders) [inline t/tick.html:7]: 😆

**Review by oalders — APPROVED** [review]
Thanks @jmadler and @simbabque! Wow, pull request #1. That was well before my time. So nice to see these little bits of history making it into the codebase.

**Comment by simbabque** [comment]
I don't think I had even heard of github in 2011. I was still hacking away using CGI and SVN.

File: t/tick.html (hunk 1 above)
Comment (jmadler) [reply t/tick.html:7]: Not a change I would have made, but I'll approve 😆

**Review by jmadler — APPROVED** [review]

**Pull request merged**